### PR TITLE
DCOS-47925: remove manual trigger for rerun script

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

- Check if this branch has a job with `head` that was triggered automatically without a manual check and ran successfully without rerunning the tests
- Check if this branch has a job with `head` that was triggered automatically without a manual check and ran successfully with rerunning a subset of tests
- Check if a `.only` in an integration test can be commited without a warning, renders the `merge` test red, but triggers the `branch` test to run reruns

## Trade-offs

- Decided that ~10min of work for not running is okay, compared to maintaining an extra check before the tests run. As always, feel free to challenge me on this 😇

### CODE REVIEW

File: scripts/ci/rerun-integration-tests.sh [hunk 1]
```diff
@@ -12,10 +11,16 @@ PROJECT_ROOT="$( cd "$( echo "${SCRIPT_PATH}" | sed s+/scripts/ci++)" && pwd )"
 
 source "$SCRIPT_PATH/../utils/test"
 source "$SCRIPT_PATH/../utils/integration-tests"
-
+set +e
```
Comment (juliangieseke) [inline scripts/ci/rerun-integration-tests.sh]: IIRC you can catch exit code from `find_files_with_debug_directives` by directly executing if condition:
```suggestion
if [ find_files_with_debug_directives "$PROJECT_ROOT/$TESTS_FOLDER" ]; then
  echo "No files found to run, aborting"
  exit 0
fi
```

File: scripts/ci/rerun-integration-tests.sh [hunk 2]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
 
-set -e
```
Comment (juliangieseke) [inline scripts/ci/rerun-integration-tests.sh]: not sure if I like 🤔
wasnt it you telling me `set -e` is basicly mandatory? 😅

File: scripts/ci/rerun-integration-tests.sh (hunk 1 above)
Comment (juliangieseke) [reply scripts/ci/rerun-integration-tests.sh]: also, as `set -e` is removed at the top of this file, this switch isnt necessary anymore, isnt it? 🤔

File: scripts/ci/rerun-integration-tests.sh (hunk 2 above)
Comment (DanielMSchmidt) [reply scripts/ci/rerun-integration-tests.sh]: I re-set it, later on before the test runs start

File: scripts/ci/rerun-integration-tests.sh (hunk 1 above)
Comment (DanielMSchmidt) [reply scripts/ci/rerun-integration-tests.sh]: I see, I inlined the subshell and that also worked without the `+e`, so I'm using that now

**Pull request merged**